code42/py42
# Feature/add trusted domain methods

### Description of Change ###

- New `trustedactivities` client with the following new methods:
    - `sdk.trustedactivities.get_all()` to get all trusted activities.
    - `sdk.trustedactivities.create()` to create a new trusted activity.
    - `sdk.trustedactivities.get()` to get details about a single trusted activity.
    - `sdk.trustedactivities.update()` to update a trusted activity.
    - `sdk.trustedactivities.delete()` to delete a trusted activity

- New custom exceptions
    - `Py42TrustedActivityInvalidChangeError`
    - `Py42TrustedActivityConflictError`
    - `Py42TrustedActivityInvalidCharacterError`

### Testing Procedure ###
Users should be able to use the py42 sdk methods of the `trustedactivities` client to hit the corresponding API endpoints to add, update, delete, and retrieve trusted activity details.  Currently the two types of activity supported are `DOMAIN` (urls) and `SLACK` (slack workspaces).  

Unit tests for the client and services can be found in `tests/clients/test_trustedactivities.py` and `tests/clients/test_trustedactivities.py`, respectively.  A separate story specifies adding integration tests and adding the service to the mock-servers.

### PR Checklist ###
Did you remember to do the below?
